Common/src/main/java/com/blamejared/crafttweaker/api/util/sequence/SequenceManager.java

```diff
@@ -0,0 +1,91 @@
+package com.blamejared.crafttweaker.api.util.sequence;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Manages {@link Sequence}s, handles adding, removing and ticking.
+ */
+public class SequenceManager {
+    
+    private static final Map<SequenceType, List<Sequence<?, ?>>> sequences = new HashMap<>();
+    private static final Map<SequenceType, List<Sequence<?, ?>>> sequencesView = Collections.unmodifiableMap(sequences);
+    
+    /**
+     * Adds a new Sequence.
+     *
+     * @param type     The type of Sequence to add.
+     * @param sequence The sequence to add.
+     */
+    public static void addSequence(SequenceType type, Sequence<?, ?> sequence) {
+        
+        sequences.computeIfAbsent(type, sequenceType -> new LinkedList<>())
+                .add(sequence);
+    }
+    
+    /**
+     * Removes a sequence.
+     *
+     * @param type     The type of Sequence to remove.
+     * @param sequence The sequence to remove.
+     */
+    public static void removeSequence(SequenceType type, Sequence<?, ?> sequence) {
+        
+        sequences.getOrDefault(type, Collections.emptyList())
+                .remove(sequence);
+    }
+    
+    /**
+     * Tick all sequences for the given type on the given side.
```

Comment:
Outdated docs, the side part is now part of the sequenceType so it's no longer necessary to write about it here